### PR TITLE
fix: ViewRenderer

### PR DESCRIPTION
**Pull request by jakharbek**
| Q             | A
| ------------- | ---
| Is bugfix?    | yes/no
| New feature?  | yes/no
| Breaks BC?    | yes/no
| Tests pass?   | yes/no
| Fixed issues  | comma-separated list of tickets # fixed by the PR, if any

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -19,6 +19,10 @@
         {
             "name": "Alexander Makarov",
             "email": "sam@rmcreative.ru"
+        },
+        {
+            "name": "Javharbek Abdulatipov",
+            "email": "jakharbek@gmail.com"
```
Comment (rustamwin) [inline composer.json:25]: the "authors" section should be removed.

File: src/Extensions/Yii_Twig_Extension.php [hunk 2]
```diff
@@ -0,0 +1,61 @@
+<?php
+/**
+ * @link http://www.yiiframework.com/
+ * @copyright Copyright (c) 2008 Yii Software LLC
+ * @license http://www.yiiframework.com/license/
```
Comment (rustamwin) [inline src/Extensions/Yii_Twig_Extension.php:5]: This doc block should be removed.

File: src/Extensions/Yii_Twig_Extension.php [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+/**
+ * Class Yii_Twig_Extension
+ * @package Yiisoft\Yii\Twig\Extensions
+ */
+class Yii_Twig_Extension extends \Twig\Extension\AbstractExtension implements GlobalsInterface
+{
+    /**
+     * @var ContainerInterface
+     */
+    private $container;
```
Comment (rustamwin) [inline src/Extensions/Yii_Twig_Extension.php:22]:
```suggestion
    private ContainerInterface $container;
```

File: src/Extensions/Yii_Twig_Extension.php [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+     */
+    public function __construct(ContainerInterface $container)
+    {
+        $this->container = $container;
+    }
+
+    /**
+     * @return array|\Twig\TwigFunction[]
+     */
+    public function getFunctions()
```
Comment (rustamwin) [inline src/Extensions/Yii_Twig_Extension.php:36]: typehint required.

File: src/ViewRenderer.php [hunk 5]
```diff
@@ -7,346 +7,67 @@
...
 /**
- * TwigViewRenderer allows you to use Twig templates in views.
+ * Class ViewRenderer
+ * @package Yiisoft\Yii\Twig
  *
- * @property array $lexerOptions @see self::$lexerOptions. This property is write-only.
- *
- * @author Alexander Makarov <sam@rmcreative.ru>
- * @since 2.0
+ * @author Javharbek Abdulatipov <jakharbek@gmail.com>
```
Comment (rustamwin) [inline src/ViewRenderer.php:22]: should be removed.

File: src/ViewRenderer.php [hunk 6]
```diff
@@ -7,346 +7,67 @@
...
-        $this->addExtensions([new Extension($this->uses)]);
-
-        // Adding custom extensions
-        if (!empty($this->extensions)) {
-            $this->addExtensions($this->extensions);
-        }
-
-        $this->twig->addGlobal('app', Yii::getApp());
-    }
+    public $container;
```
Comment (rustamwin) [inline src/ViewRenderer.php:29]:
```suggestion
    private ContainerInterface $container;
```

File: composer.json [hunk 7]
```diff
@@ -28,15 +32,13 @@
         "yiisoft/yii-core": "^3.0@dev",
```
Comment (samdark) [inline composer.json:32]: There's no such package anymore. Should be removed.

**Comment by samdark** [comment]
[Twig](https://trello.com/c/hIPjFpYd/12-twig)

**Comment by jakharbek** [comment]
Made a change, added tests

**Comment by samdark** [comment]
That's a good start.

**Pull request merged**